Lodgify.Payments.Stripe.Server.IntegrationTests/Shared/BaseIntegrationTest.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using CacheCow.Client;
+using Lodgify.Payments.Stripe.Server.IntegrationTests.Factories;
+using Lodgify.Payments.Stripe.Server.IntegrationTests.Helpers;
+using Microsoft.EntityFrameworkCore;
+using WireMock.Admin.Mappings;
+using WireMock.Client;
+using Xunit;
+
+namespace Lodgify.Payments.Stripe.Server.IntegrationTests.Shared;
+
+[CollectionDefinition(nameof(PaymentsCollection))]
+public class PaymentsCollection : ICollectionFixture<CustomWebApplicationFactory>;
```

Comment:
It's related to the previous comments.
What test collections/fixtures we'd like to have? One for all or multiply more granular?
If the second option then how about:
```suggestion
public class AccountManagmentCollection : ICollectionFixture<CustomWebApplicationFactory>;
```

Reply:
Hmmm not really get the point here... to be sure we are on the same page the FixtureCollection simply share context among the tests.  It makes sense to have multiple collections if you need to share different contexts or setups among different groups of tests. Each collection can have its own setup and teardown logic, which can be useful for isolating tests that require different configurations or dependencies.  For example, you might have one collection for tests that require a real Stripe setup and another for tests that require another configuration. This helps in organizing tests and ensuring that the setup and teardown logic is only applied to the relevant tests, improving test isolation and reducing potential side effects.
So to sum up, personally I don`t think so we want be so granullar and create separate collection per stripe operation, my idea was to have one fixture collection for mocks and one for real connection with stripe. In this second option we can then do stuff like:

```
public class StripeApiFixture : WebApplicationFactory<Program>, IDisposable
{
    protected override void ConfigureWebHost(IWebHostBuilder builder)
    {
        builder.ConfigureServices(services =>
        {
            // Configure test services here
        });
    }

    public void Dispose()
    {
        // Clean up stripe resources
    }
}
```

What you think?

Reply:
I see. I think it is related to the previous topic. 
How we want to design our tests in terms of collections/parallel execution?

My way of thinking was to run tests related to us calling Stripe for update/create Account with one collection (AccountApi / AccountManagment), tests related to Stripe calling us to update Account through webhook with other collection etc.

I am afraid that we may develop the solution that forces us to have single base class with single fixture and all tests will be executed in the same collection because it will be too much work to introduce new collection.

Reply:
No, no it will be splitted on few collection, let`s move it to the scope of next task (real stripe integration) just to make real example of it